manifests/rules/out/dnsserver.pp

```diff
@@ -0,0 +1,45 @@
+#
+# @summary manage outgoing DNS responses from a DNS server
+#
+# @param dns_servers optional list of local ip addresses from the DNS server
+#
+class nftables::rules::out::dnsserver (
+  Array[Stdlib::IP::Address] $dns_servers = [],
+) {
+  unless empty($dns_servers) {
+    $dns_servers.each |$index,$dns| {
+      nftables::rule {
+        "default_out-dnsserverudp-${index}":
+      }
+      if $dns =~ /:/ {
+        Nftables::Rule["default_out-dnsserverudp-${index}"] {
+          content => "ip6 daddr ${dns} udp sport 53 accept",
+        }
+      } else {
+        Nftables::Rule["default_out-dnsserverudp-${index}"] {
+          content => "ip daddr ${dns} udp sport 53 accept",
+        }
+      }
+
+      nftables::rule {
+        "default_out-dnsservertcp-${index}":
+      }
```

Comment:
I guess it follow a structure as seen in other classes, but that feels quite convoluted (altering a resource that appears only after in the manifest) and I would have expected something like this that feels more simple:

```puppet
      $content = $dns ? {
        Stdlib::IP::Address::V6 => "ip6 daddr ${dns} udp sport 53 accept",
        Stdlib::IP::Address::V4 => "ip daddr ${dns} udp sport 53 accept",
      }

      nftables::rule { "default_out-dnsservertcp-${index}":
        content => $content,
      }
```

I have not checked all other rules classes, maybe such a construct make sense in some cases.  But I am not sure it is better to follow that pattern when a simpler construct can be used?

Reply:
I am not a fan of this contruct either, but I copied it from the rules::out::dns class and wanted to stay consistent.